jsplumb-persistence-plugin.js

```diff
@@ -1,207 +1,208 @@
-        connection = plumbInstance.getAllConnections();
-        var blocks = [];
-        $(options.selector).each(function(idx, elem) {
-            var $elem = $(elem);
-            var id = $elem.attr('id');
-            blocks.push({
-                id: $elem.attr('id'),
-                left: parseInt($elem.css("left"), 10),
-                top: parseInt($elem.css("top"), 10),
-                width: parseInt($elem.css("width"), 10),
-                heigth: parseInt($elem.css("heigth"), 10),
-                html: $elem.html()
-            });
-        });
-        var connections = [];
-        for (var i = 0; i < connection.length; i++) {
-            var id = connection[i].sourceId;
-            var endpoints = plumbInstance.getEndpoints(connection[i].sourceId);
-            var connector = connection[i].getConnector();
-            var type = connector.type;
-            var attrs = {};
-            switch (type) {
-                case "Bezier":
-                    attrs["curviness"] = connector.getCurviness();
-                    break;
-                case "Straight":
-                    {
-                        break;
-                    }
-                case "Flowchart ":
-                    {
-                        break;
-                    }
-                case "State Machine":
-                    {
-                        attrs["curviness"] = connector.getCurviness();
-                        break;
-                    }
-            }
-            var endpointArray = [];
-            connection[i].endpoints.forEach(function(endpoint) {
-                var options = {};
-                if (endpoint.type == 'Image')
-                {
-                    options.url = endpoint.canvas.src;
-                    options.anchor = endpoint.anchor;
-                    endpointArray.push([endpoint.type, options]);
-                } else
-                {
-                    options.anchor = endpoint.anchor;
-                    endpointArray.push([endpoint.type, options]);
-                }
-                console.log(options);
-            });
+  jsPlumbInstance.save = function(options,plumbInstance)
+  {
+      if(!options || !options.selector){
+          return {};
+      }
+      plumbInstance = plumbInstance || jsPlumb;
+      var connection;
+      connection = plumbInstance.getAllConnections();
+      var blocks = [];
+      $(options.selector).each(function(idx, elem) {
+          var $elem = $(elem);
+          var id = $elem.attr('id');
+          blocks.push({
+              id: $elem.attr('id'),
+              left: parseInt($elem.css("left"), 10),
+              top: parseInt($elem.css("top"), 10),
+              width: parseInt($elem.css("width"), 10),
+              heigth: parseInt($elem.css("heigth"), 10),
+              html: $elem.html()
+          });
+      });
+      var connections = [];
+      $.each(instance.getAllConnections(), function (idx, connection) {
```

Comment:
`instance` appears to be global variable which will not be present
We should be saving using the `jsPlumbInstance` passed to the save call.